cub/test/catch2_test_device_reduce.cuh

```diff
@@ -437,6 +437,60 @@ void compute_segmented_argmax_reference(const thrust::device_vector<item_t> &d_i
   }
 }
 
+/**
+ * @brief Helper function to compute the reference solution for unique keys (i.e., collapsing each
+ * run of equal keys into a single key).
+ */
+template <typename in_it_t, typename out_it_t>
+inline out_it_t compute_unique_keys_reference(in_it_t h_in_begin,
+                                              std::size_t num_keys,
+                                              out_it_t h_out_it)
+{
+  if (num_keys == 0)
+  {
+    return h_out_it;
+  }
+  *h_out_it++ = h_in_begin[0];
+  for (std::size_t i = 1; i < num_keys; i++)
+  {
+    if (!(h_in_begin[i - 1] == h_in_begin[i]))
+    {
+      *h_out_it = h_in_begin[i];
+      h_out_it++;
+    }
+  }
+  return h_out_it;
+}
+
+/**
+ * @brief Helper function to compute the reference solution for unique keys (i.e., collapsing each
+ * run of equal keys into a single key).
+ */
+template <typename item_t>
```

Comment:
Will stop suggesting the change, please apply our style guide (CamelCase for template parameters) to the rest of the places.
```suggestion
template <typename ItemT>
```